modules/dcache/src/main/java/diskCacheV111/util/Pgpass.java

```diff
@@ -75,51 +79,93 @@ private boolean parseUrl(String url) {
-            PrintWriter stdOutput = new PrintWriter(new BufferedWriter(new OutputStreamWriter(p1.getOutputStream())));
-            String reply = stdInput.readLine();
-            try {
-                p1.waitFor();
-            }
-            catch (InterruptedException x) {
-                System.out.println("stat for '"+_pwdfile+"' was interrupted");
-                stdInput.close(); stdError.close(); stdOutput.close();
-                return null;
-            }
-            stdInput.close(); stdError.close(); stdOutput.close();
+            if (checkPgFilePermissions("rw-------")) {
+                result = parsePgFile(hostname, port, database, username);
 
-            if (reply==null) {
-                System.out.println("Cannot stat '"+_pwdfile+"'");
+            } else {
+                System.out.println("Protection for '" + _pwdfile + "' must be '600'");
                 return null;
-            } else if (!reply.equals("'600'")) {
-                System.out.println("Protection for '"+_pwdfile+"' must be '600'");
-                return null;
-            }
-            /*
-             * Here we can read and parse the password file
-             */
-            try {
-                BufferedReader in = new BufferedReader(new FileReader(_pwdfile));
-                String line, r = null;
-                while ((line = in.readLine()) != null && r == null) {
-                    r = process(line, hostname, port, database, username);
-                }
-                in.close();
-                return r;
-            } catch (IOException e) {
-                System.out.println("'"+_pwdfile+"': I/O error");
-                return null;
-            }
 
+            }
+        } catch (IOException e) {
+            System.out.println("'" + _pwdfile + "': I/O error");
+            return null;
         }
-        catch (IOException ex) {
-            System.out.println("Cannot stat "+_pwdfile);
+
+        return result;
+    }
+
+    /**
+     * Check if the pwd file exists
+     *
+     * @return exist? then true
+     */
+    private boolean checkIfFileExists() {
+        return new File(_pwdfile).isFile();
+    }
+
+    /**
+     * Check the pwd file for selectable permissions. The file must be a POSIX file, at the moment.
+     *
+     * @param referencePermissionInput The permissions the file should have. It's in the unix like format.
+     *                                 e.g. "rwx------" for owner read-write-execute
+     * @return permissions right? true
+     * @throws IOException
+     */
+    private boolean checkPgFilePermissions(String referencePermissionInput) throws IOException {
+
+        if(checkIfOsIsPosixCompliant()){
+            return checkPgFilePermissionsForPosix(referencePermissionInput);
+
+        } else {
+            System.out.println("Error reading permissions for '" + _pwdfile + "'. OS is not POSIX compliant");
```

Comment:
Use slf4j to log errors, not `println`.

You should add a static field member, something like:

    private static final Logger LOGGER = LoggerFactory.getLogger("diskCacheV111.util.Pgpass");